src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Administration/GeneralDataProvider.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\Administration;
+
+use Cookie;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use PrestaShopBundle\Form\Exception\DataProviderError;
+use PrestaShopBundle\Form\Exception\DataProviderErrorCollection;
+use PrestaShopBundle\Form\Exception\DataProviderException;
+
+/**
+ * This class is responsible of managing the data manipulated using general form
+ * in "Configure > Advanced Parameters > Administration" page.
+ */
+final class GeneralDataProvider implements FormDataProviderInterface
+{
+    /**
+     * If you set cookie lifetime value too high there can be multiple problems.
+     * Hours are converted to seconds, so int might be turned to float if it's way to high.
+     * Cookie classes crashes if lifetime goes beyond year 9999, there are probably multiple other things.
+     * So we need to set some sort of max value. 100 years seems like a lifetime beyond reasonable use.
```

Comment:
> 100 years seems like a lifetime beyond reasonable use

yes 📘 😄  if a shop powered by PrestaShop is still running in 100 years, we can say this code was a success !